src/ui/src/app/shared/services/deployments.service.ts

```diff
@@ -50,7 +50,79 @@ export class DeploymentsService {
 
   private extractData(res: Response, fallback = {}) {
     let body = res.json();
-    return body.data || fallback;
+    var data = body.data
+    if (!data) {
+        return fallback
+    }
+    var attributes = data.attributes
+    if (attributes) {
+      attributes.urls = []
+      var resources = attributes.resources
+      if (resources) {
+        var parsedResources = this.parseResources(resources)
+        var svc = parsedResources['v1/Service']
+        if (svc) {
+          svc.forEach(x => this.svcToURLs(x, attributes.urls))
+        }
+      }
+    }
+    return data || fallback;
+  }
+
+  /**
+   * Take a service status and try to assemble urls out of it
+   * @param portSpec parsed SVC line
+   * @param ret out parameter to store urls in 
+   */
+  private svcToURLs(portSpec, ret) {
+    // pattern is INT_PORT:EXT_PORT/TCP
+    const portsRE = /^(\d+:)?(\d+)\/TCP$/
+    // match ip4/6 ips, as opposed to <pending>
+    const ipRE = /^[\d\.:]+$/
+    const EXT_IP = "EXTERNAL-IP"
+
+    var ports = portSpec['PORT(S)'].split(",")
+    ports.forEach(port => {
+      var extIP = portSpec[EXT_IP]
+      // only look at services with valid external IP
+      if (!ipRE.exec(extIP))
+        return
+      var portMatch = portsRE.exec(port)
+      if (portMatch) {
+        var protocol = portMatch[1] == '443:' ? 'https' : 'http'
+        ret.push(`${protocol}://${extIP}:${portMatch[2]}`)
+      }
+    })
+  }
+
+  /**
+   * Parse resource string and return sections as tables
+   * @param {string} resourceStr 'helm status' output
+   * @returns object mapping section to array of contents 
+   */
+  private parseResources(resourceStr) {
```

Comment:
Any chance we could use/extend/replace the parser that is part of https://github.com/helm/monocular/blob/master/src/ui/src/app/deployment/deployment.component.ts#L57 ?